documentation/optee_design.md

```diff
@@ -615,53 +614,64 @@ and are built along with and statically built into the OP-TEE core blob.
 
 The pseudo Trusted Applications included in OP-TEE already are OP-TEE
 secure privileged level services hidden behind a "GlobalPlatform TA Client" API.
-These pseudo-TAs are used for various purpose as specific secure services or
-embedded tests services.
+These pseudo-TAs are used for various purposes such as specific secure services
+or embedded tests services.
 
 Pseudo TAs do not benefit from the GlobalPlatform Core Internal API support
 specified by the GlobalPlatform TEE specs. These APIs are provided to TAs as a
 static library each TA shall link against (the "libutee") and that calls OP-TEE
-core service through system calls. As OP-TEE core does link with the
-libutee, Pseudo TAs can only use the OP-TEE core internal APIs and
-routines.
+core service through system calls. As OP-TEE core does not link with libutee,
+Pseudo TAs can only use the OP-TEE core internal APIs and routines.
 
 As pseudo TAs have the same privileged execution level as the OP-TEE core code
 itself, such situation may not be desirable for complex TAs.
 
-In most cases a real, dynamically loaded TA is the best choice instead of adding
+In most cases an unprivileged (user mode) TA is the best choice instead of adding
 your code directly to the OP-TEE core.  However if you decide your application
 is best handled directly in OP-TEE core like this, you can look at
 `core/arch/arm/pta/stats.c` as a template and just add your pseudo TA based on
 that to the `sub.mk` in the same directory.
 
-### Trusted Applications
+### User Mode Trusted Applications
 
-Trusted Applications (TAs) are applications dynamically loaded by OP-TEE
+User Mode Trusted Applications are loaded (mapped into memory) by OP-TEE
 core in the Secure World when something in the REE wants to talk to that
-particular application UUID.  It is similar to the way the Linux
-kernel can dynamically load kernel modules, although unlike with Linux, in
-OP-TEE TAs actually run at a lower CPU privileged level than OP-TEE core code.
-
-Because the TAs are signed by the same key that built the OP-TEE core, they
-are able to be stored in the untrusted REE filesystem, and tee-supplicant will
-take care of passing them to be checked and loaded by the Secure World OP-TEE
-core.  Again this is simular to Linux kernel module signature checking.
+particular application UUID. They run at a lower CPU privilege level
+than OP-TEE core code. In that respect, they are quite similar to regular
+applications running in the Rich Execution Environment (REE), except that
+they execute in Secure World.
 
 Trusted Application benefit from the GlobalPlatform Core Internal API as
 specified by the GlobalPlatform TEE specifications.
 
-Trusted Application consist of a cleartext signed ELF file, named from the UUID
+There are two types of user mode TAs, which differ by the way they are stored.
```

Comment:
proposal:
* There are **~~two~~several** types of user mode TAs, which differ by the way they are stored **~~.~~**
**in the system and get loaded in the execution context created by the OP-TEE core.**

Reply:
At the moment there are only two types